### DISCUSSION TITLE
My IP seems to be blocked by GitHub

### DISCUSSION THREAD

**Discussion by LeeLin2602** [community/community#57075] [Other Feature Feedback, Questions, & Ideas] [answered]
<img width="810" alt="image" src="https://github.com/community/community/assets/30391069/b223bf0c-f776-4271-858b-11af2042c779">

When I send a request to https://github.com, it just hangs there.

My IP is 103.179.29.10.

**Comment by DSDmark** [comment]
Hi @LeeLin2602,


I think, you mean to send request on `https://github.com`

As you can see, the server responds with an **HTTP 301** status code, along with a **Location** header that specifies the new location (the HTTPS URL). This tells the client (your browser or curl) to resend the request to the HTTPS version.

```sh
curl -v https://github.com
```

When you use `http://github.com/` you are making an unsecured HTTP request. However, GitHub has likely implemented a redirection policy to enforce the use of HTTPS for security reasons. This means that when you send an **HTTP request**, GitHub automatically redirects you to the secure HTTPS version.

**Reply by LeeLin2602** [reply]
I tried to send request on http://github.com, as you said, it gave me a reponse with 301 status code. It works indeed.

But when I try to send request on https://github.com, it just hangs there. That's the problem I would like to ask.

Thank you for replying and indicating the misunderstanding statement I made.

**Reply by DSDmark** [reply]
@LeeLin2602, Now, can you show us what error is show on your terminal when you made that `https://github.com` change.

**Reply by LeeLin2602** [reply]
I post an image above. There's no error message, only timeout during tls handshake.
It's ok to curl http://github.com and it responds, but when curl https://github.com, it just timeout.

**Reply by DSDmark** [reply]
Hmm... it is an issue, establishing a connection using the **OpenSSL** `SSL_ERROR_SYSCALL` which is a low-level system call. I search on internet and found that There could be several reasons:

- Connectivity or network issues
- Firewall or proxy restrictions
- Certificate verification (This could happen if your system's certificate store is outdated or misconfigured)

You can try to check your internet connect or try to switch different network then try again or try another browser may also help to figure out your problem and check your firewall settings to make sure that the port 443 (HTTPS) is not blocked. And check if OpenSSL is not installed correctly

**Reply by airtower-luna** [reply]
A certificate verification error would look very different, but a network/firewall issue seems likely. You could try using `traceroute` (preferably with TCP packets) to figure out *where* your packets get dropped.

**Comment by LeeLin2602** [comment, marked as answer]
I solve this problem via increasing MTU from 1200 to 1500.
Indeed, this is a network issue.
